docker/buildx.sh

```diff
@@ -21,59 +21,65 @@ EXAMPLES
 REFERENCE
     https://github.com/openziti/ziti-tunnel-sdk-c/blob/main/docker/BUILD.md
 EOF
-    [[ $# -eq 1 ]] && {
-        return $1
-    } || {
+    if [[ $# -eq 1 ]]; then
+        return "$1"
+    else
         return 0
-    }
+    fi
 }
 
-BASENAME=$(basename $0) || exit $?
-DIRNAME=$(dirname $0) || exit $?
+#BASENAME=$(basename $0) || exit $?
+DIRNAME=$(dirname "$0") || exit $?
 
-while getopts :r:c OPT;do
+while getopts :r:chl OPT;do
     case $OPT in
         r)  CONTAINER_REPO=$OPTARG 
             ;;
         c) 	FLAGS+=$OPT     # don't checkout vZITI_VERSION
             ;;
         h) _usage; exit 0   # not an error
             ;;
-        *|\?) _usage 1      # error
+        l)  FLAGS+=$OPT     # also tag and push latest
+            ;;
+        \?|*) _usage 1      # error
             ;;
     esac
 done
 shift "$((OPTIND-1))"
 
 # default to latest
-: ${ZITI_VERSION:=$(git fetch --quiet --tags && git tag -l|sort -Vr|head -1|sed -E 's/^v(.*)/\1/')}
+: "${ZITI_VERSION:=$(git fetch --quiet --tags && git tag -l|sort -Vr|head -1|sed -E 's/^v(.*)/\1/')}"
 
 # required opts
-[[ ! -z ${CONTAINER_REPO:+true} ]] || {
+if [[ -z "${CONTAINER_REPO:-}" ]]; then
     echo "ERROR: missing -r REPO option to define container image repository name for image push" >&2
     _usage; exit 1
-}
+else
+    TAG_PARAMS="--tag=\"${CONTAINER_REPO}:${ZITI_VERSION}\""
+fi
 
 if [[ ${FLAGS:-} =~ c ]]; then
     echo "WARN: not checking out Git tag v${ZITI_VERSION}"
 else
     git diff --exit-code # bail if unstaged differences
     git fetch --tags
-    git checkout v${ZITI_VERSION}
+    git checkout "v${ZITI_VERSION}"
 fi
 
-docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
-egrep -q enabled /proc/sys/fs/binfmt_misc/qemu-arm
-docker run --rm arm64v8/alpine uname -a|egrep -q 'aarch64 Linux'
-docker run --rm arm32v7/alpine uname -a|egrep -q 'armv7l Linux'
+if [[ ${FLAGS:-} =~ l ]]; then
+    TAG_PARAMS+=" --tag=\"${CONTAINER_REPO}:latest\""
+fi
 
+docker run --rm --privileged docker/binfmt:a7996909642ee92942dcd6cff44b9b95f08dad64
+grep -E -q 'enabled' /proc/sys/fs/binfmt_misc/qemu-arm
+docker run --rm --platform linux/arm64/v8 arm64v8/alpine uname -a |grep -Eq 'aarch64 Linux'
+docker run --rm --platform linux/arm/v7 arm32v7/alpine uname -a|grep -Eq 'armv7l Linux'
 docker buildx create --use --name=ziti-builder 2>/dev/null || docker buildx use --default ziti-builder
 
-docker buildx build $DIRNAME \
-    --platform linux/amd64,linux/arm/v7,linux/arm64 \
-    --build-arg ZITI_VERSION=${ZITI_VERSION} \
-    --tag ${CONTAINER_REPO}:${ZITI_VERSION} \
-    --tag ${CONTAINER_REPO}:latest \
+eval docker buildx build "${DIRNAME}" \
```

Comment:
Why use `eval` here? Does the buildx output include a shell command that this shell is executing?